### PR TITLE
Create entity fix

### PR DESCRIPTION
**Pull request by juddruckers**
Fix for bug that would cause makeGetDenormalizedData return an empty object even if the entity exists and the feature slice status is in a successful state. 

Add makeGetIsSliceSuccessful that checks if the feature slice state is in one of two successful states HAS_FETCHED_SUCCESS or HAS_CREATED_SUCCESS

use makeGetIsSliceSuccessful in makeGetDenormalizedData to prevent it from returning an empty object if the feature slice status is in HAS_CREATED_SUCCESS. 

Add test to check makeGetDenormalizedData returns empty object if feature slice status is in not in a successful state.

### CODE REVIEW

File: src/featureSelectors.ts [hunk 1]
```diff
@@ -60,6 +60,17 @@ export const makeGetHasEntityCreatedSuccess = ({ name }: { name: string }) => {
   );
 };
 
+export const makeGetIsSliceSuccessful = ({ name }: { name: string }) => {
```
Comment (gkadillak) [inline src/featureSelectors.ts:63]: I'm not a huge fan of this selector because it means that if another status of HAS_*_SUCCESS is created, they have to remember to add that value here which is a potential bug. Instead, I think it's better to check if there is a status at all. We shouldn't actually care what it is. That means that this selector goes away and line 95 will check to see if the status is truthy.

File: src/featureSelectors.ts [hunk 2]
```diff
@@ -76,12 +87,12 @@ export const makeGetDenormalizedData = ({
...
+  const getIsSliceSuccessful = makeGetIsSliceSuccessful({ name });
   return createSelector(
     getFeatureSlices,
-    getHasFeatureFetchedSuccess,
-    (slices: FeatureSlices<FeatureSlice>, hasFetchedSuccessfully: boolean) => {
-      if (!hasFetchedSuccessfully || !slices.entities) {
+    getIsSliceSuccessful,
+    (slices: FeatureSlices<FeatureSlice>, getIsSliceSuccessful: boolean) => {
+      if (!getIsSliceSuccessful || !slices.entities) {
         return {};
```
Comment (gkadillak) [inline src/featureSelectors.ts:96]: maybe this should return undefined? Not sure why I was returning an empty object here

File: src/featureSelectors.test.ts [hunk 3]
```diff
@@ -29,13 +29,40 @@ describe("featureSelectors", () => {
...
+
       expect(output).toEqual({
         users: [
           { id: 1, name: "Garrett" },
           { id: 2, name: "Laura" },
         ],
       });
     });
+
+    it("should return an empty object", () => {
```
Comment (gkadillak) [inline src/featureSelectors.test.ts:43]: the test name here should be more description to say what you're trying to test. why should it return an empty object? What condition is this testing?

File: src/featureSelectors.test.ts [hunk 4]
```diff
@@ -29,13 +29,40 @@ describe("featureSelectors", () => {
...
+        },
+      };
+      const getDenormalizedData = makeGetDenormalizedData({
+        name: "users",
+        entity: "users",
+      });
+
+      const output = getDenormalizedData.resultFunc(featureSlices, true);
+
+      expect(output).toEqual(undefined);
```
Comment (gkadillak) [inline src/featureSelectors.test.ts:65]: please use [`.toBeFalsy`](https://jestjs.io/docs/expect#tobefalsy) here

File: src/featureSelectors.test.ts [hunk 5]
```diff
@@ -29,13 +29,40 @@ describe("featureSelectors", () => {
...
+          data: { animals: [1, 2] },
+          status: "foo",
+        },
+      };
+      const getDenormalizedData = makeGetDenormalizedData({
+        name: "users",
+        entity: "users",
+      });
+
+      const output = getDenormalizedData.resultFunc(featureSlices, true);
```
Comment (gkadillak) [inline src/featureSelectors.test.ts:63]: please use a constant for each value passed to the selector. it helps improve the readability of the selector, especially when you're using `resultFunc`

File: src/featureSelectors.ts [hunk 6]
```diff
@@ -76,13 +76,11 @@ export const makeGetDenormalizedData = ({
...
 }) => {
-  const getHasFeatureFetchedSuccess = makeGetHasFeatureFetchedSuccess({ name });
   return createSelector(
     getFeatureSlices,
-    getHasFeatureFetchedSuccess,
-    (slices: FeatureSlices<FeatureSlice>, hasFetchedSuccessfully: boolean) => {
-      if (!hasFetchedSuccessfully || !slices.entities) {
-        return {};
+    (slices: FeatureSlices<FeatureSlice>) => {
+      if (!slices[name].status|| !slices.entities) {
```
Comment (gkadillak) [inline src/featureSelectors.ts:82]: please put `slices[name].status` in a variable

File: src/featureSelectors.test.ts [hunk 7]
```diff
@@ -29,13 +29,40 @@ describe("featureSelectors", () => {
...
+
       expect(output).toEqual({
         users: [
           { id: 1, name: "Garrett" },
           { id: 2, name: "Laura" },
         ],
       });
     });
+
+    it("should return undefined if the selected slice has a falsy status", () => {
```
Comment (gkadillak) [inline src/featureSelectors.test.ts:43]: Good titles for tests should tell the user what they're testing without being too specific. Here's a title that I think would be appropriate: "should return falsy value when the status of the slice is not yet defined"

**Comment by gkadillak** [comment]
also, @juddruckers, please make sure to double check that this is still functioning correctly in another project

**Review by gkadillak-lyra — APPROVED** [review]
LGTM!
